recipes/proposal/all/test_package/conanfile.py

```diff
@@ -1,17 +1,25 @@
-from conans import ConanFile, CMake, tools
 import os
-
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"
+    requires = "spdlog/1.9.2" # TODO: not sure why this is explicitly necessary, but it fails without it
```

Comment:
Conan 2.0 changes the graph model and deps are private by default to encourage good software practices.

you need to add a `transitive_header=True` with a link to the reason why it's need (ie a comment from the bot failing is perfect or a link to the source code)

Reply:
I am honestly note sure why I would need to put `transitive_headers=True` for spdlog, but not for any other libraries. 
If I don't (and do not explicitly write `requires = "spdlog/1.9.2"` in the conanfile of the test_package), I get the error
```
/usr/local/include/spdlog/fmt/fmt.h:27:14: fatal error: fmt/core.h: No such file or directory
   27 | #    include <fmt/core.h>
```
which sounds related to the experience in #11981 and #15832...

I could do this, and this does fix the issue, but I would like to understand why this is necessary 😅